openssl-dynamic/src/main/java/io/netty/tcnative/jni/ocsp/OcspCallback.java

```diff
@@ -0,0 +1,160 @@
+ *          78:1a:e7:3d:a2:f3:c0:2c:cb:d0:1d:a9:ae:bb:e4:c7:c0:f2:
+ *          89:98:de:fa
+ * ======================================
+ * </code>
+ *
+ * @see SSLContext#isOcspAvailable()
+ * @see SSLContext#newOcspCallback(long, boolean, OcspCallback)
+ * @see SSLContext#freeOcspCallback(long, OcspCallback)
+ * @see SSL#setOcspStaple(long, byte[])
+ * @see SSL#getOcspStaple(long)
+ */
+public interface OcspCallback {
+
+    // Extracted from https://github.com/openssl/openssl/blob/master/include/openssl/tls1.h
+    
+    /**
+     * SERVER ONLY: The server has an OCSP staple and used {@link SSL#setOcspStaple(long, byte[])}
+     * to set it.
+     * 
+     * NOTE: The user must call {@link SSL#setOcspStaple(long, byte[])} prior to returning this value.
+     * 
+     * SSL_TLSEXT_ERR_OK
+     */
+    public static final int OCSP_SERVER_ACK = 0;
+
+    /**
+     * SERVER ONLY: The server did not have an OCSP staple or chose to not provide one and would like
+     * to proceed without one. This is effectively the default behavior if OCSP stapling wasn't enabled
+     * and verification is left up to the client.
+     * 
+     * SSL_TLSEXT_ERR_ALERT_WARNING
+     */
+    public static final int OCSP_SERVER_NACK = 1;
+    
+    /**
+     * SERVER ONLY: Some error occurred. Throwing an {@link Exception} from the {@link #callback(long)}
+     * method will have the same effect.
+     * 
+     * SSL_TLSEXT_ERR_ALERT_FATAL
+     */
+    public static final int OCSP_SERVER_ERROR = 2;
+    
+    /**
+     * SERVER ONLY: The server may or may not have an OCSP staple but instead of letting the handshake
+     * proceed it'd like to abort it. The reasons could be that the server has determined on its own that
+     * the certificate has been revoked or the server was unable to obtain the OCSP staple from the CA
+     * and it's deeming that it's unsafe to let the client follow the default behavior.
+     * 
+     * NOTE: OpenSSL's own documentation on this status code is a bit misleading.
+     * 
+     * SSL_TLSEXT_ERR_NOACK
+     */
+    public static final int OCSP_SERVER_REJECT = 3;
+    
+    /**
+     * CLIENT ONLY: Some error occurred. Throwing an {@link Exception} from 
+     * the {@link #callback(long)} method will have the same effect.
+     */
+    public static final int OCSP_CLIENT_ERROR = -1;
+
+    /**
+     * CLIENT ONLY: The OCSP staple obtained via {@link SSL#getOcspStaple(long)} is not good 
+     * and the SSL connection will fail.
+     */
+    public static final int OCSP_CLIENT_NACK = 0;
+
+    /**
+     * CLIENT ONLY: The OCSP staple obtained via {@link SSL#getOcspStaple(long)} is good.
+     */
+    public static final int OCSP_CLIENT_ACK = 1;
+
+    /**
+     * Called by OpenSSL for each new {@link SSLEngine} that has OCSP stapling enabled.
+     * 
+     * @see SSL#setOcspEnabled(long, boolean)
+     * @see SSLContext#newOcspCallback(long, boolean, OcspCallback)
+     */
+    public int callback(long ssl) throws Exception;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Avoid modifiers which are implied by the context [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUnusedModifier)